CRM/Event/Form/SelfSvcTransfer.php

```diff
@@ -363,54 +386,44 @@ public function postProcess() {
       $value_to['is_pay_later'] = $dao->is_pay_later;
       $value_to['fee_amount'] = $dao->fee_amount;
     }
-    $value_to['contact_id'] = $contact_id;
-    $value_to['event_id'] = $this->_event_id;
+    $value_to['contact_id'] = $toContactID;
+    $value_to['event_id'] = CRM_Core_DAO::getFieldValue('CRM_Event_BAO_Participant', $fromParticipantID, 'event_id');
     $value_to['status_id'] = 1;
     $value_to['register_date'] = date("Y-m-d");
     //first create the new participant row -don't set registered_by yet or email won't be sent
     $participant = CRM_Event_BAO_Participant::create($value_to);
     //send a confirmation email to the new participant
-    $this->participantTransfer($participant);
+    self::sendConfirmationEmail($participant);
     //now update registered_by_id
     $query = "UPDATE civicrm_participant cp SET cp.registered_by_id = %1 WHERE  cp.id = ({$participant->id})";
-    $params = [1 => [$this->_from_participant_id, 'Integer']];
+    $params = [1 => [$fromParticipantID, 'Integer']];
     $dao = CRM_Core_DAO::executeQuery($query, $params);
     //copy line items to new participant
-    $line_items = CRM_Price_BAO_LineItem::getLineItems($this->_from_participant_id);
-    foreach ($line_items as $item) {
+    $line_items = CRM_Price_BAO_LineItem::getLineItems($fromParticipantID);
+    foreach ($line_items as $id => $item) {
+      //Remove contribution id from older participant line item.
+      CRM_Core_DAO::singleValueQuery("UPDATE civicrm_line_item SET contribution_id = NULL WHERE id = %1", [1 => [$id, 'Integer']]);
```

Comment:
this is the only change needed for this ticket to resolve. Rest is the separation of code into a single function to pass the unit test assertion.

Reply:
The code should not be changing financial stuff just through what appears in the UI by changing the line item table, UNLESS that is the only error. I haven't looked carefully at this, but can you confirm that the financial entries in civicrm_financial_item, civicrm_financial_trxn and civicrm_entity_financial_trxn are being update correctly. 

IIRC the registration is supposed to move to a new contact, but the contribution information is supposed to be unaffected.

If the contribution is being transferred as well as the participant registration, or if it is being cancelled, then there should be entries for the first contribution that reverse out the effect of the relevant line item, and a new contribution and line item should be added to the contact receiving the registration. But I don't think this is the case. The $10 contribution should be left alone, unchanged.